src/cascadia/TerminalApp/CascadiaSettings.cpp

```diff
@@ -151,6 +151,55 @@ void CascadiaSettings::_CreateDefaultProfiles()
     _profiles.emplace_back(powershellProfile);
     _profiles.emplace_back(cmdProfile);
     
+    // Add all WSL distros from: HKCU\Software\Microsoft\Windows\CurrentVersion\Lxss
+    Profile WSLDistro{};
+    HKEY hLXSSKey;
+    // Should I use RegOpenCurrentUser to access the uncached profile?
+    // https://docs.microsoft.com/en-us/windows/desktop/api/winreg/nf-winreg-regopencurrentuser#remarks
+    if (RegOpenKeyExW(HKEY_CURRENT_USER, L"Software\\Microsoft\\Windows\\CurrentVersion\\Lxss",
+        0, KEY_READ, &hLXSSKey) == ERROR_SUCCESS)
+    {
+        std::filesystem::path AppsDirectory = ExpandEnvironmentVariableString(L"%ProgramFiles%");
+        AppsDirectory /= "WindowsApps";
+        std::filesystem::path WSLCommandLine = L"";
+
+        wchar_t WSLKey[MAX_PATH] = L""; // Registry key for a distinct WSL Distribution
+        wchar_t DistributionName[MAX_PATH] = L""; // The name to be displayed by the flyout list
+        wchar_t PackageFamilyName[MAX_PATH] = L""; // The distinct name for the distribution
+
+        // These tell the Registry functions how much space they have to work with when returning a value.
+        DWORD WSLKey_Length = MAX_PATH;
+        DWORD DistributionName_Length = MAX_PATH;
+        DWORD PackageFamilyName_Length = MAX_PATH;
+
+        for (int i = 0;
+            RegEnumKeyExW(hLXSSKey, i, WSLKey, &WSLKey_Length, NULL, NULL, NULL, NULL) == ERROR_SUCCESS;
+            i++)
+        {
+            // Reset available space for each value.
+            WSLKey_Length = MAX_PATH;
+            DistributionName_Length = MAX_PATH;
+            PackageFamilyName_Length = MAX_PATH;
+
+            if (RegGetValueW(hLXSSKey, WSLKey, L"DistributionName", RRF_RT_REG_SZ, NULL,
+                DistributionName, &DistributionName_Length) == ERROR_SUCCESS

```

Comment:
I don't know what the Terminal project's guidelines are around line length but splitting a multi-conditional across lines in this fashion may be challenging to read for some readers.  As such, consider either indenting the continuation or not wrapping the invocation to RegGetValueW across multiple lines.

Reply:
I've since been told they would like to keep line length around 100 characters.